perses/annihilation/relative.py

```diff
@@ -237,13 +237,20 @@ def __init__(self,
         # TODO: Refactor this into self._add_particles()
         _logger.info("Adding and mapping old atoms to hybrid system...")
         for particle_idx in range(self._topology_proposal.n_atoms_old):
-            particle_mass = self._old_system.getParticleMass(particle_idx)
+            particle_mass_old = self._old_system.getParticleMass(particle_idx)
+            
+            if particle_idx in self._topology_proposal.old_to_new_atom_map.keys():
+                particle_index_in_new_system = self._topology_proposal.old_to_new_atom_map[particle_idx]
+                particle_mass_new = self._new_system.getParticleMass(particle_index_in_new_system)
+                particle_mass = (particle_mass_old + particle_mass_new) / 2 # Take the average of the masses if the atom is mapped
+            else:
+                particle_mass = particle_mass_old
```

Comment:
This looks fine to me. I have the feeling one could simplify the logic here, specially considering the if statement in line 253. But I guess this is easy enough to follow as it is and the possible performance gained by it is not critical, so this is not a blocker. Thanks!!